src/jit/importer.cpp

```diff
@@ -10779,6 +10779,10 @@ MATH_MAYBE_CALL_NO_OVF:         ovfl = false;
 
             if  ((op1->gtFlags & GTF_SIDE_EFFECT) || opts.compDbgCode)
             {
+#if FEATURE_MULTIREG_STRUCT_RET
```

Comment:
As we discussed offline, we don't need to materialize an assignment if op1 is a non-helper call not marked with GTF_CALL_M_RETBUFFARG.  

The reason, we have to exclude helper calls is that helpers that has struct return val don't follow Linux ABI and always accept by ref as a hidden argument irrespective of the struct size.